version.py

```diff
@@ -40,18 +40,27 @@
 
 
 def call_git_describe(prefix='', cwd='.'):
-    cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    version_name_cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    last_tag_cmd = 'git describe --tags --abbrev=0'
     try:
         version = subprocess.run(
-            cmd.split(),
+            version_name_cmd.split(),
             stdout=subprocess.PIPE,
             check=True,
         ).stdout.decode()
+
+        last_tag = subprocess.run(
+            last_tag_cmd.split(),
+            stdout=subprocess.PIPE,
+            check=True,
+        ).stdout.decode()
+
         version = version.strip()[len(prefix):]
-        if '-' in version:
-            version = '{}+dev{}.{}'.format(*version.split('-'))
+        last_tag = last_tag.strip()[len(prefix):]
+        if version != last_tag:
+            version = f"{last_tag}+dev{version[len(last_tag)+1:].replace('-','.')}"
```

Comment:
Looks good but I would like to see some comment why it's done this way. I.e. that it helps with tags that have hyphens in them like described in an issue.

Reply:
I see that there's an assumption that `version_name_cmd` is either equal to `last_tag_cmd` or contains it as a prefix. Adding an `assert` for this assumption would make that more apparent to someone reading the code and also, conveniently, let Python check and and alert us in development if your assumption does not hold true some case.

Reply:
I did this this way because we want to support all versioning so I needed to get last tag name, and name of tag where present code is and then see if there is any difference - if is add prefix `+dev` beetwen last tag name and number commits behind this tag.

I didn't know where you want comment - here - or in code so if you want this in code I will rewrite it. :)

Reply:
@jiivan Do you want some comment in code about how it's done or here?

Reply:
Also added asset so @cameel part Done

Reply:
In code

Reply:
Done